shared/settings/advanced/index.tsx

```diff
@@ -208,6 +250,152 @@ class Developer extends React.Component<Props, State> {
   }
 }
 
+// A list so the order of the elements is fixed
+const ProxyTypeToDisplayNameList = [
+  ['noProxy', 'No Proxy'],
+  ['httpConnect', 'HTTP Connect'],
+  ['socks', 'Socks5'],
+]
+const ProxyTypeToDisplayName = ProxyTypeToDisplayNameList.reduce((obj, item) => {
```

Comment:
i'd invert this relationship. just have a map of the short name to long name and have the array be the order of the short names, aka

```
const proxyTypeOrder = ['noProxy', 'httpConnect'...
const proxyTypeToDisplayName = invert(displayNameToProxyType)
```

Reply:
Done.